### ISSUE TITLE
Cannot using k8s external ip address in the remote_servers.xml configuration?

### ISSUE THREAD

**Issue by chaos827** [Altinity/clickhouse-operator#1551]
I created a 2 * 2 ClickHouse DB (2 Shards and each Shard has 2 replicas) using ClickHouse-Operator and azure Kubernetes(AKS), meanwhile I created load balance services for each replica (each pod has its own load balance service and unique external IP), it works well.
After that I updated the remote_servers xml file using the external ip instead of hostname, in this way the distribute query (i.e: create database TestDB on CLUSTER '{cluster}'  ENGINE = Atomic) is not worked on the pod which with external  ip,  also the ReplicatedMergreeTree is not synced the data in the same pod, but the pod work well when I used HostName or Pod Ip, below is my remote_servers configuration in the yaml,

config.d/remote_servers.xml: 
        <yandex>          
          <remote_servers>
              
              <testcluster>
                  <shard>
                      <internal_replication>true</internal_replication>
                      <replica>
                          <host>chi-p1-testcluster-0-0-0.chi-p1-testcluster-0-0.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                      <replica>
                          <host>chi-p1-testcluster-0-1-0.chi-p1-testcluster-0-1.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                     
                  </shard>
                  <shard>
                      <internal_replication>true</internal_replication>
                      <replica>
                          <host>chi-p1-testcluster-1-0-0.chi-p1-testcluster-1-0.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                      <replica>
                          <host>10.224.0.192</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                    
                  </shard>
              </testcluster>
              
              <all-replicated>
                  <shard>
                      <internal_replication>true</internal_replication>
                      <replica>
                          <host>chi-p1-testcluster-0-0-0.chi-p1-testcluster-0-0.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                      <replica>
                          <host>chi-p1-testcluster-0-1-0.chi-p1-testcluster-0-1.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                      <replica>
                          <host>chi-p1-testcluster-1-0-0.chi-p1-testcluster-1-0.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                      <replica>
                          <host>10.224.0.192</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                     
                  </shard>
              </all-replicated>
              <all-sharded>
                  <shard>
                      <internal_replication>false</internal_replication>
                      <replica>
                          <host>chi-p1-testcluster-0-0-0.chi-p1-testcluster-0-0.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                  </shard>
                  <shard>
                      <internal_replication>false</internal_replication>
                      <replica>
                          <host>chi-p1-testcluster-0-1-0.chi-p1-testcluster-0-1.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                  </shard>
                  <shard>
                      <internal_replication>false</internal_replication>
                      <replica>
                          <host>chi-p1-testcluster-1-0-0.chi-p1-testcluster-1-0.clickhouse1.svc.cluster.local</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                  </shard>
                  <shard>
                      <internal_replication>false</internal_replication>
                      <replica>
                          <host>10.224.0.192</host>
                          <port>9000</port>
                          <user>test</user>
                          <password>test123</password>
                          <secure>0</secure>
                      </replica>
                  </shard>                  
                  
              </all-sharded>
          </remote_servers>
        </yandex>

I did this test beacuse I want to set up the ClickHouse in the different data center (replica1 in primary and replica2 in the geolocation), so I have to split the ClickHouse in the two AKS, and using external ip to communicate, but I do not undershand why my yaml is not work,  does someone know the root causes?
many thanks!

**Comment by UnamedRus** [comment]
It's 2 different problems.
Replication doesn't care about remote_servers configuration, you need to check/fix interserver_http_host parameter instead. (and make sure that port 9009 is exposed)

https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings#interserver-http-host

**Comment by chaos827** [comment]
hi @UnamedRus, thank you for sharing the suggestion. I updated my yaml to config the interserver_http_host parameter, however I tried serval ways, but the replica is still not work, this my new part in the yaml
config.d/interserver_http_host.xml:
        <yandex>
          <interserver_http_host>10.225.0.10(load balancer external ip)</interserver_http_host>
          <interserver_http_post>9009</interserver_http_post>
          
          <tcp_port>9000</tcp_port>
          <interserver_http_credentials>
            <user>test</user>
            <password>test123</password>
          </interserver_http_credentials>
        </yandex>
also I double confirmed the port 9009 already exposed

**Comment by chaos827** [comment]
yes it is weird, because I want to set up ClickHouse cross region (cross AKS), the pod IP is dynamic, so I have to create Load Balance service for each replica.

**Comment by alex-zaitsev** [comment]
See https://altinity.com/blog/setting-up-cross-region-clickhouse-replication-in-kubernetes
